PyDynamic/model_estimation/fit_filter.py

```diff
@@ -87,92 +86,207 @@ def _fitIIR(
     return b, a
 
 
-def LSIIR(Hvals, Nb, Na, f, Fs, tau=0, justFit=False):
-    """Least-squares IIR filter fit to a given frequency response.
-    
-    This method uses Gauss-Newton non-linear optimization and pole
-    mapping for filter stabilization
-    
+def LSIIR(
+    Hvals: np.ndarray,
+    Nb: int,
+    Na: int,
+    f: np.ndarray,
+    Fs: float,
+    tau: Optional[int] = 0,
+    justFit: Optional[bool] = False,
+    verbose: Optional[bool] = True,
+    max_stab_iter: Optional[int] = 50,
+    inv: Optional[bool] = False,
+    UHvals: Optional[np.ndarray] = None,
+    mc_runs: Optional[int] = 1000,
+) -> Union[
+    Tuple[np.ndarray, np.ndarray, int], Tuple[np.ndarray, np.ndarray, int, np.ndarray]
+]:
+    """Least-squares (time-discrete) IIR filter fit to a given frequency response or
+    its reciprocal
+
+    For the forward problem this method uses Gauss-Newton non-linear optimization.
+    For the inverse problem it uses the equation-error method. The filter then is
+    stabilized by pole mapping and introduction of a time delay. Associated
+    uncertainties are optionally propagated when provided using the Monte Carlo method.
+
     Parameters
     ----------
-        Hvals:   numpy array of (complex) frequency response values of shape (M,)
-        Nb:      integer numerator polynomial order
-        Na:      integer denominator polynomial order
-        f:       numpy array of frequencies at which Hvals is given of shape
-        (M,)
-        Fs:      sampling frequency
-        tau:     integer initial estimate of time delay
-        justFit: boolean, when true then no stabilization is carried out
-    
+    Hvals : array_like of shape (M,)
+        (Complex) frequency response values.
+    Nb : int
+        Order of IIR numerator polynomial.
+    Na : int
+        Order of IIR denominator polynomial.
+    f : array_like of shape (M,)
+        Frequencies at which `Hvals` is given.
+    Fs : float
+        Sampling frequency for digital IIR filter.
+    tau : int, optional
+        Initial estimate of time delay for filter stabilization (default = 0). If
+        `justFit = True` this parameter is not used and `tau = 0` will be returned.
+    justFit : bool, optional
+        If True then no stabilization is carried out, if False (default) filter is
+        stabilized. This parameter is only available for reasons of backward
+        compatibility. If you intend to skip stabilization please from now on use
+        `max_stab_iter = 0` and leave `justFit` untouched. `justFit` will be removed
+        in a future release.
```

Comment:
Let's take the parameter out, mention the deprecation of it and provide the solution straight away.

Reply:
Resolved in 472dc0b6b27c2d974c3558f5f275c6c8aa211bc4.